src/Microsoft.Extensions.Logging.Abstractions/Internal/TypeNameHelper.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.Extensions.Logging.Abstractions.Internal
+{
+    public class TypeNameHelper
```

Comment:
@Eilon I hope I understood your comment right...I copied this and its tests from Common repo and modified here to suit our needs